### PR TITLE
Allow 3 column layouts to use refresh instead of postMessage

### LINKED ISSUE #28: Three column layouts can't appear in Customizer dynamically
The `tertiary` sidebar cannot appear dynamically in the Customizer because it is conditionally loaded in the `sidebar-tertiary.php` template partial on page load.

So if you are using a one- or two-column layout when the Customizer is initially loaded, the `tertiary` sidebar will not appear upon selecting a three-column layout.

Possible solutions:

A. Always render the sidebar content and show/hide it with CSS.

B. Inject the sidebar into the DOM via the Customizer controls JS.

### PR DESCRIPTION
**Pull request by jonathanbardo**
@fjarrett Please test & review

This should fix #28.

We are also exposing a filter for child themes to specify is a layout needs refresh or not, thus making this flexible.

### CODE REVIEW

**Comment by tobeycodes** [comment]
@jonathanbardo Not sure what you think but it probably makes sense to use postMessage for full_width as well and just toggle the no-max-width class.

**Comment by jonathanbardo** [comment]
@schrapel We certainly could. Let's create a separate issue for this.

**Comment by tobeycodes** [comment]
Also this isn't working for me Chrome 52 / Mac. No console errors and the file is enqueued properly on the latest version. I'll try debug

**Comment by jonathanbardo** [comment]
@schrapel It works for me on chrome 52 on mac. Are you testing using a child theme or on primer directly? @fjarrett Can you check? Might be a caching issue of some kind.

**Comment by tobeycodes** [comment]
It was primer directly. I checked the contents of layouts.js being loaded
and tried hard refreshing. I got as far as checking when it runs. It would
run the function on the initial load but it didn't appear to when I changed
the layout. The actual layout changed but didn't refresh with the 3rd column

It may be a issue at my end though but I'm off the computer right now

**Comment by frankiejarrett** [comment]
@jonathanbardo @schrapel Working fine for me on Chrome/Mac.

File: inc/customizer/controls/layouts.php [hunk 1]
```diff
@@ -12,7 +12,43 @@ public function enqueue() {
...
+		$layouts['three-column-reversed'] = 'refresh';
+
+		/**
+		 * Filter layouts transport mechanism in the customizer. Either postMessage or refresh.
+		 *
+		 * @since 1.0.0
+		 *
+		 * @var array
+		 */
+		$layouts = apply_filters( 'primer_layouts_transport', $layouts );
```
Comment (frankiejarrett) [inline inc/customizer/controls/layouts.php]: @jonathanbardo As a best-practice we should type cast filter results. Done in https://github.com/godaddy/wp-primer-theme/commit/4165462e0b03c815bc79cb5fa4b2161d0c68d7e1

**Pull request merged**